services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/HankeServiceITests.kt

```diff
@@ -931,6 +930,36 @@ class HankeServiceITests : DatabaseTest() {
         verify { applicationService.create(any(), USER_NAME) }
     }
 
+    @Test
+    fun `Hanke alku and loppu calculated from alueet`() {
+        val hanke = getATestHanke()
+        val a = DateFactory.getEndDatetime()
+        val b = DateFactory.getEndDatetime().plusMonths(1)
+        val c = DateFactory.getEndDatetime().plusMonths(2)
+        val d = DateFactory.getEndDatetime().plusMonths(3)
+        hanke.alueet[0].haittaAlkuPvm = a
+        hanke.alueet[0].haittaLoppuPvm = c
+        val hankealue =
+            HankealueFactory.create(
+                haittaAlkuPvm = b,
+                haittaLoppuPvm = d,
+                kaistaHaitta = TodennakoinenHaittaPaaAjoRatojenKaistajarjestelyihin.KOLME,
+                kaistaPituusHaitta = KaistajarjestelynPituus.KOLME,
+                meluHaitta = Haitta13.KOLME,
+                polyHaitta = Haitta13.KOLME,
+                tarinaHaitta = Haitta13.KOLME,
+            )
+        hanke.alueet.add(hankealue)
+        val createdHanke = hankeService.createHanke(hanke)
```

Comment:
Creating the hanke in the service doesn't affect this test. This test tests that the `alkuPvm` and `loppuPvm` fields are read from the hankealueet of the hanke. It's not really a service test, and certainly not an integration test. The test is useful, but it should be somewhere in unit tests, like in `HankeTest`.

Reply:
Moved to HankeTest.